Loop/Managers/DeviceDataManager.swift

```diff
@@ -397,6 +397,8 @@ final class DeviceDataManager {
                     }
                 }
             }
+        case .unreliableData:
```

Comment:
If we aren't returning CGM value (that happen to be unreliable), how is this any different than the `.noData` rules? I don't see why we could cancel a high-temp in the case of unreliable data and not missing data.

Or perhaps the unreliable data event occurs *after* getting new data as an indicator that "the previously received CGM data is now believed to be unreliable" and that is why we are taking action to cancel any high-temp?

Furthermore, do we need some indicator that the data is now reliable again. Won't the next loop just re-enact the high-temp that was just cancelled unless there is some state that CGM data is in an unreliable state?

Reply:
@darinkrauss Yes, these are CGM states that call into question previous values, as the sensor algorithm is now saying the current values aren't coherent enough to actually show a value. If Loop is high temping off of a high bg reading, and we have a missed reading, it's very likely that you still need the high temp, as Loop calculates the high temp rate to delivery the needed insulin for that high bg reading over 30m. Ideally, this state wouldn't actually completely nullify the last dosing decision, but would rather decrease confidence in the forecast, which might call for a new dosing decision that's not necessarily scheduled basal. But we don't have a confidence based forecast at this point.

Reply:
> Furthermore, do we need some indicator that the data is now reliable again. Won't the next loop just re-enact the high-temp that was just cancelled unless there is some state that CGM data is in an unreliable state?

The actual presence of a new BG value that triggers a new loop is the signal.  Loop won't loop if there is no new BG data.